util/src/main/java/com/google/cloud/hadoop/util/RetryHttpInitializer.java

```diff
@@ -226,14 +231,16 @@ public RetryHttpInitializer(
       String defaultUserAgent,
       int maxRequestRetries,
       int connectTimeoutMillis,
-      int readTimeoutMillis) {
+      int readTimeoutMillis,
+      Map<String, String> headers) {
```

Comment:
This constructor seems too big, let's create an option class with builder and pass it in constructor instead.

Reply:
This seems like a fairly big refactor. I will have a separate PR to create this option class.

Reply:
Good point.